scripts/make_submission_file.py

```diff
@@ -45,54 +51,71 @@ def write_is2re_relaxations(paths, filename, hybrid):
             submission_file[f"{split}_ids"] = np.array(ids)
             submission_file[f"{split}_energy"] = np.array(energies)
 
-    np.savez_compressed(filename, **submission_file)
+    np.savez_compressed(args.out_path, **submission_file)
 
 
-def write_predictions(paths, filename):
-    submission_file = {}
+def write_predictions(args, dataset):
+    if args.is2re_relaxations:
+        write_is2re_relaxations(args, dataset=dataset)
+    else:
+        submission_file = {}
+
+        for split in SPLITS[dataset]:
+            res = np.load(vars(args)[split], allow_pickle=True)
+            verify_dtype(res, dataset)
+            contents = res.files
+            for i in contents:
+                key = "_".join([split, i])
+                submission_file[key] = res[i]
+
+        np.savez_compressed(args.out_path, **submission_file)
 
-    for idx, split in enumerate(["id", "ood_ads", "ood_cat", "ood_both"]):
-        res = np.load(paths[idx], allow_pickle=True)
-        contents = res.files
-        for i in contents:
-            key = "_".join([split, i])
-            submission_file[key] = res[i]
 
-    np.savez_compressed(filename, **submission_file)
+def verify_dtype(preds, dataset):
+    if dataset == "OC22":
+        if "energy" in preds:
+            assert preds["energy"].dtype in [
+                np.float32,
+                np.float64,
+            ], "Predictions written in the wrong precision. Ensure `total_energy` flag is True in the config."
+        if "forces" in preds:
+            assert preds["forces"].dtype in [
+                np.float32,
+                np.float64,
+            ], "Predictions written in the wrong precision. Ensure `total_energy` flag is True in the config."
```

Comment:
I'd drop this from here; this script to package a submission file doesn't seem like the right place to assert dtypes (as long as it's all np.float).

Reply:
I dropped this function in favor of adding `config["task"]["dataset"] == "oc22_lmdb"` to the if statement — this should guarantee that OC22 predictions are written in float32 because that flag is required to load the OC22 dataset. Let me know if there is an issue with this choice. https://github.com/Open-Catalyst-Project/ocp/blob/2d21fd9c8e1836a28c375963a201d0a7eb11a156/ocpmodels/trainers/forces_trainer.py#L217-L221